src/Orchard/DisplayManagement/Descriptors/ShapePlacementStrategy/ShapePlacementParsingStrategy.cs

```diff
@@ -111,8 +111,9 @@ public class ShapePlacementParsingStrategy : IShapeTableProvider {
             differentiator = "";
             shapeType = shapeLocation.ShapeType;
             var dashIndex = shapeType.LastIndexOf('-');
-            if (dashIndex > 0 && dashIndex < shapeType.Length - 1) {
-                differentiator = shapeType.Substring(dashIndex + 1);
+            var dashIndex = shapeType.LastIndexOf("__");

```

Comment:
Will this compile? you have dashIndex specified twice?

            var dashIndex = shapeType.LastIndexOf('-');
            var dashIndex = shapeType.LastIndexOf("__");